test-count.js

```diff
@@ -0,0 +1,327 @@
+      testCountTemplate(testCount, packageName, "manual");
+    }
+
+    if (lineText.includes(smoke)) {
+      testCountTemplate(testCount, packageName, "smoke");
+    }
+
+    if (lineText.includes(smoke) && lineText.includes(todo)) {
+      testCountTemplate(testCount, packageName, "smoke_todo");
+    }
+
+    if (lineText.includes(regres)) {
+      testCountTemplate(testCount, packageName, "regres");
+    }
+
+    if (lineText.includes(regres) && lineText.includes(todo)) {
+      testCountTemplate(testCount, packageName, "regres_todo");
+    }
+
+    if (
+      lineText.includes(broken) ||
+      lineText.includes(unverified) ||
+      lineText.includes(precondition)
+    ) {
+      if (lineText.includes(todo)) {
+        testCountTemplate(testCount, packageName, "other_todo");
+      } else {
+        testCountTemplate(testCount, packageName, "other");
+      }
+    }
+  }
+}
+
+function fileParser(fileName, directoryPath) {
+  let universalTag;
+  let packageName;
+  const directoryElements = directoryPath.split("/");
+  testCount.set(`${fileName}_line_no`, 0);
+
+  var r = readline.createInterface({
+    input: fs.createReadStream(directoryPath + "/" + fileName),
+  });
+
+  r.on("line", function (text) {
+    if (testCount.get(`${fileName}_line_no`) === 0) {
+      if (text.includes("@")) {
+        packageName = text.trim().split("@")[1].trim();
+        if (tags.includes(packageName)) {
+          packageName =
+            directoryElements[directoryElements.indexOf("packages") + 1].trim();
+        }
+      } else {
+        packageName =
+          directoryElements[directoryElements.indexOf("packages") + 1].trim();
+      }
+
+      if (packageName === "dev-console") {
+        packageName =
+          directoryElements[directoryElements.indexOf("features") + 1].trim();
+      }
+
+      if (!packages.includes(packageName)) {
+        packages.push(packageName);
+      }
+
+      let filteredArray = text.split("@").filter((word) => {
+        if (tags.includes(word)) return word;
+      }); // it is assumend that only one tag is filtered
+      universalTag = filteredArray[0];
+    }
+    lineParser(text, packageName, universalTag);
+
+    testCount.set(
+      `${fileName}_line_no`,
+      testCount.get(`${fileName}_line_no` + 1)
+    );
+  });
+
+  r.on("close", () => {
+    if (completedFiles.get(`${fileName}`) === undefined) {
+      completedFiles.set(`${fileName}`, 1);
+    } else {
+      formattedOutput();
+    }
+  });
+}
+
+async function crawlDirectory(directoryPath) {
+  if (fs.lstatSync(directoryPath).isDirectory()) {
+    fs.readdir(directoryPath, (err, files) => {
+      if (err) {
+        throw err;
+      } else {
+        files.forEach((file) => {
+          if (
+            path.extname(file) === ".feature" &&
+            !directoryPath.includes("e2e")
+          ) {
+            fileParser(file, directoryPath);
+          }
+          if (!file.includes(".") && fs.statSync(directoryPath)) {
```

Comment:
`fs.statSync` returns an object so the right side of the condition will never be false. What was the intention here?